sdk/python/feast/feature.py

```diff
@@ -62,3 +63,86 @@ def from_proto(cls, feature_proto: FeatureProto):
         feature.update_shape_type(feature_proto)
         feature.update_domain_info(feature_proto)
         return feature
+
+
+class FeatureRef:
+    """ Feature Reference represents a reference to a specific feature.  """
+
+    def __init__(self, name: str, feature_set_name: str = None):
+        self.proto = FeatureRefProto(name=name, feature_set_name=feature_set_name)
+
+    @classmethod
+    def from_proto(cls, proto: FeatureRefProto):
+        """
+        Construct a feature reference from the given FeatureReference proto
+
+        Arg:
+            proto: Protobuf FeatureReference to construct from
+
+        Returns:
+            FeatureRef that refers to the given feature
+        """
+        return cls(name=proto.name, feature_set_name=proto.feature_set_name)
+
+    @classmethod
+    def from_str(cls, feature_ref_str: str, ignore_project: bool = False):
+        """
+        Parse the given string feature reference into FeatureRef model
+        String feature reference should be in the format feature_set:feature.
+        Where "feature_set" and "name" are the feature_set name and feature name
+        respectively.
+
+        Args:
+            feature_ref_str: String representation of the feature reference
+            ignore_project: Ignore projects in given string feature reference
+                            instead throwing and error
+
+        Returns:
+            FeatureRef that refers to the given feature
+        """
+        proto = FeatureRefProto()
+        if "/" in feature_ref_str:
+            if ignore_project:
+                _, feature_ref_str = feature_ref_str.split("/")
+            else:
+                raise ValueError(f"Unsupported feature reference: {feature_ref_str}")
+
+        # parse feature set name if specified
+        if ":" in feature_ref_str:
+            proto.feature_set_name, feature_ref_str = feature_ref_str.split(":")
```

Comment:
Same comment as above. I think it would be clearer if we didn't reuse the `feature_ref_str` name and instead used something like `feature_name`

Reply:
The `feature_ref_str` reuse is imposed  by the flexibility of feature references. I have used this pattern in various  string Feature Reference parsing methods throughout the codebase, not just in the python SDK